lib/scholar/cluster/k_means.ex

```diff
@@ -0,0 +1,325 @@
+defmodule Scholar.Cluster.KMeans do
+  @moduledoc """
+  K-Means algorithm
+  """
+  import Nx.Defn
+  import Scholar.Shared
+
+  @derive {Nx.Container, containers: [:num_iterations, :clusters, :inertia, :labels]}
+  defstruct [:num_iterations, :clusters, :inertia, :labels]
+
+  @doc """
+  Fits a K-Means model for sample inputs `x`
+
+
+  ## Options
+
+    * `:num_clusters` - The number of clusters to form as well as the number of centroids to generate. Defaults to `8`.
```

Comment:
I would make `num_clusters` required. Also, perhaps this doc could emphasize this is the `K` in `K-means`

Reply:
Yes, I've been thinking about requiring num_clusters. In sci-kit learn 8 is the default, but the required parameter also makes sense.